### PR TITLE
Add colorizaion of rules events to the table printer

### PR DESCRIPTION
**Pull request by AsafEitani**
## Initial Checklist

- [ ] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)
Adds colorization to the table printer on rules events.

commit bec6f3670a4f34c9e41b2b686a548332721cf43e (HEAD -> color_table, origin/color_table)
Author: AsafEitani <eitaniasaf@gmail.com>
Date:   Mon Dec 19 20:35:31 2022 +0200

    printer: add colorizaion of rules events

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

### CODE REVIEW

File: pkg/cmd/printer/printer.go [hunk 1]
```diff
@@ -127,6 +128,14 @@ func (p tableEventPrinter) Print(event trace.Event) {
 	if p.relativeTS {
 		ut = ut.UTC()
 	}
+
+	isRule := false
+	for _, v := range events.Definitions.Get(events.ID(event.EventID)).Sets {
+		if v == "rules" {
+			isRule = true
+		}
+	}
```
Comment (yanivagman) [inline pkg/cmd/printer/printer.go:137]: Instead, you can simply check if ID is between StartRulesID to MaxRulesID

**Comment by rafaeldtinoco** [comment]
A change like this should be optional (set by a cmdline argument). There might be other tools parsing tracee output... having ANSI escape codes will disturb those. GNU tools usually have something like "--color=auto" or "--color=always" (or =no).

**Comment by AsafEitani** [comment]
> A change like this should be optional (set by a cmdline argument). There might be other tools parsing tracee output... having ANSI escape codes will disturb those. GNU tools usually have something like "--color=auto" or "--color=always" (or =no).

Do you think that users parse the table printer? not using JSON?

**Comment by rafaeldtinoco** [comment]
> Do you think that users parse the table printer? not using JSON?

TBH, I don't. But it is also a bad practice to colorize without explicitly requesting, or checking the TERMCAP support (imagine the user is running this in a vt100 terminal, or in any other terminal type that does not support colors).

**Comment by josedonizetti** [comment]
@AsafEitani I like the idea of colors, and think it should be enabled by default. But also agree with @rafaeldtinoco and the user should be able to disable it. Can we go with the same solution `trivy` uses? The library `https://github.com/fatih/color` allows us to use color, and the user can set a env var to disable it, eg: `NO_COLOR=true` and unset it, if he wants color back. 

WDYT?

If you want to see an example on trivy -> https://github.com/aquasecurity/trivy/blob/main/pkg/report/table/table.go#L22

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
Per last comments.

**Comment by AsafEitani** [comment]
Waiting for #2541 to implement the color by severity

**Comment by rafaeldtinoco** [comment]
@AsafEitani, https://github.com/aquasecurity/tracee/issues/2355 (took care of #2541) is finished. Do you really think we should colorize the output ? @josedonizetti opinions ? Maybe to have another printer (now that we support multiple printers) for the colorized output and derived from the regular table printer ? Just thinking out loud.

Converting this to draft for now.

**Comment by AsafEitani** [comment]
I don't have a strong opinion, what ever you choose.

**Comment by rafaeldtinoco** [comment]
Lets close this for now then. We can re-discuss if we ever think there is a need for it.

**Pull request closed without merging**